### PR TITLE
fix(kit): remove `HostBinding` in `input-inline`

### LINKED ISSUE #1220: 🐞 - data-value doesn't detect changes on TuiInputInlineComponent
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/taiga-starter-wmnqh2

### Description

data-value doesn't detect changes case changeDetection: ChangeDetectionStrategy.OnPush

### Angular version

12.2.4

### Taiga UI version

2.26.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1220

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/151378473-5477c994-8081-4bc5-b3ef-24d0bc62efd6.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by splincode** [comment]
@waterplea please, review me

**Comment by splincode** [comment]
@waterplea @vladimirpotekhin please, review me

File: projects/cdk/abstract/interactive.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export abstract class AbstractTuiInteractive {
 
     focusVisible = false;
 
-    constructor() {
+    protected constructor() {
```
Comment (waterplea) [inline projects/cdk/abstract/interactive.ts:63]: I remember seeing some issues with this approach, but maybe I'm mistaken. @vladimirpotekhin @MarsiBarsi do you remember anything in that regard?

File: projects/kit/components/input-inline/input-inline.component.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ import {TUI_VALUE_ACCESSOR_PROVIDER} from '@taiga-ui/kit/providers';
 })
 export class TuiInputInlineComponent
     extends AbstractTuiControl<string | number>
-    implements TuiFocusableElementAccessor
+    implements TuiFocusableElementAccessor, OnInit
```
Comment (waterplea) [inline projects/kit/components/input-inline/input-inline.component.ts:40]: Oops

**Review by waterplea — APPROVED** [review]

File: projects/cdk/abstract/interactive.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/abstract/interactive.ts:63]: image

Abstract classes still can't be called directly without inheritance

Comment (vladimirpotekhin) [reply projects/cdk/abstract/interactive.ts:63]: I don't remember

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/kit/components/input-inline/input-inline.component.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/components/input-inline/input-inline.component.ts]: fixed

**Pull request merged**